src/targets/python/python3.js

```diff
@@ -90,8 +94,14 @@ module.exports = function (source, options) {
     .push('res = conn.getresponse()')
     .push('data = res.read()')
     .blank()
-    .push('print(data.decode("utf-8"))')
-
+  
+  // Decode response
+  if (headers['accept-encoding'] == 'gzip') {
```

Comment:
This is close, but it's not quite right I'm afraid.

In this code you're checking the request headers, but that doesn't mean that the response is actually using gzip. `accept-encoding` just means that client told the server it _could_ use gzip - there's lots of cases where the server won't actually use gzip in the response despite this.

To do this correctly, we'll need to check the `content-encoding` value in the _response_ headers, at runtime, so outputting this `if` into the Python code itself. Does that make sense? I think we do only need to do that if `accept-encoding` is set (and that'll help to keep the snippets simpler in the cases where this isn't required.

Do you want to take a look at that?

Reply:
yes it makes sense to check  the `content-encoding` in the response. i will change to check for encoding in  response headers. But in the case there is no response and `accept-encoding` is gzip, we will have to decide whether to decompress the response or not?

Reply:
Ah, no, I don't mean the response in the HAR! I mean the actual response the Python code receives when you run the snippet. There will always be a response, because the Python code has to wait for the response before it does the check & decodes the body.

Basically, if there is an accept-encoding header set, then we need to output python code like:

```python
if res.headers['content-encoding'] == 'gzip':
  print(gzip.decompress(data).decode("utf-8"))
else:
  print(data.decode("utf-8"))
```

Reply:
i see. this makes more sense. i will add the check. I think the response data should still be kept while parsing the HAR file. There might be use of it in the future.

Reply:
@pimterry content-encoding check should be good now?

Reply:
Sorry for the delay. Yes - the check looks good now! I think that approach will work nicely.

But the new change does mean that we change the output for _every_ case, and it's a bit of extra noise & confusion that most requests won't need...

I think we should add this only if there is an accept-encoding request header that contains `gzip`. Does that make sense? That would keep all the other test cases the same, but add this only when we might get a gzip result. Can you add that?

Reply:
added and updated test case